transport/src/test/java/io/netty/bootstrap/BootstrapTest.java

```diff
@@ -275,6 +275,30 @@ public void testAsyncResolutionFailure() throws Exception {
         assertThat(connectFuture.channel().isOpen(), is(false));
     }
 
+    @Test
+    public void testChannelFactoryFailureNotifiesPromise() throws Exception {
+
+        final String exceptionMessage = "newChannel crash";
+
+        final Bootstrap bootstrap = new Bootstrap()
+                .handler(dummyHandler)
+                .group(groupA)
+                .channelFactory(new ChannelFactory<Channel>() {
+            @Override
+            public Channel newChannel() {
+                throw new RuntimeException(exceptionMessage);
```

Comment:
Please create the exception before the bootstrap and just throw it. This way you can use `assertSame(...)` below to check if the same exception. This is more robust compared to check for the message.